tests/test_main.py

```diff
@@ -118,9 +119,17 @@ def test_write_results_compressed(output_path):
     assert gzip.open(f_path, "rt").read() == "id\n1\n2\n"
 
 
-def test_read_dummy_data_file(tmp_path):
-    dummy_data_file = tmp_path / "dummy_data_file.csv"
-    dummy_data_file.write_text("id\n1\n2\n", encoding="utf-8")
+@pytest.mark.parametrize(
+    "context,suffix",
+    [
+        (functools.partial(open, mode="w"), ".csv"),
```

Comment:
No objection at all to `functools.partial` here, but just flagging up that you can also use `lambda:` to achieve something similar in test parameters and in some cases that can be clearer.

Reply:
Oh! Yes, nice. Thank you for your comment. I've just compared `lambda` to `functools.partial` and I think I prefer the latter, because I pass some extra kwargs in the test itself.